OS collector fails with message "failed collecting os metrics:<nil> CreateFile D:\pagefile.sys: The system cannot find the file specified."

We are using 0.19.0 version with Prometheus 2.34.0.
windows_exporter_collector_success shows 0 for os collector. All other collectors succeeded.

we checked that pagefile is indeed enabled only on d: drive with "system managed" size. It is disabled on c: drive

we already tried executing lodctr /R many times and restarting windows exporter windows service many times.
Windows exporter windows service runs in local system account.

Windows exporter config:

collectors:
  enabled: cpu,cpu_info,cs,iis,logical_disk,logon,memory,net,os,process,service,system,time

![image](https://user-images.githubusercontent.com/93990393/192543483-1205d5ed-dbb4-4a56-947d-76fce9507dc1.png)

@breed808 

Thanks for the fix. Unfortunately, the exporter now crashes:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x38 pc=0x14d239b]

goroutine 40 [running]:
github.com/prometheus-community/windows_exporter/collector.(*OSCollector).collect(0xc0000faa80, 0xc0000c65b8, 0x0?)
        /home/r718037/promu/windows_exporter/collector/os.go:209 +0x163b
github.com/prometheus-community/windows_exporter/collector.(*OSCollector).Collect(0x0?, 0x0?, 0x0?)
        /home/r718037/promu/windows_exporter/collector/os.go:136 +0x1d
main.execute({0x174ff2a, 0x2}, {0x185cfe0, 0xc0000faa80}, 0x0?, 0x0?)
        /home/r718037/promu/windows_exporter/exporter.go:200 +0x84
main.windowsCollector.Collect.func2({0x174ff2a, 0x2}, {0x185cfe0?, 0xc0000faa80?})
        /home/r718037/promu/windows_exporter/exporter.go:141 +0x9b
created by main.windowsCollector.Collect
        /home/r718037/promu/windows_exporter/exporter.go:139 +0x5c5
```

The [line](https://github.com/prometheus-community/windows_exporter/blob/fb062dc02056958746f44ee7e0cf07bce81ff56a/collector/os.go#L209)
```go
fsipf += float64(file.Size())
```
needs to be wrapped in an `else` block. If I do that, everything works as expected and the OS exporter runs without error.

--- a/collector/os.go
+++ b/collector/os.go
@@ -202,10 +202,12 @@ func (c *OSCollector) collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) (
 	for _, pagingFile := range pagingFiles {
 		fileString := strings.ReplaceAll(pagingFile, `\??\`, "")
 		file, err := os.Stat(fileString)
+		// For unknown reasons, Windows doesn't always create a page file. Continue collection rather than aborting.
 		if err != nil {
-			return nil, err
+			log.Debugf("Failed to read page file (reason: %s): %s\n", err, fileString)
+		} else {
+			fsipf += float64(file.Size())
 		}
-		fsipf += float64(file.Size())
 	}
 
 	gpi, err := psapi.GetPerformanceInfo()

